### PR TITLE
ZonalStatus fix, generic errors in mec services for missing mandatory properties, updated rnis rab_info query params

### PR DESCRIPTION
**Pull request by pastorsx**
1) ZonalStatus AP threshold fix (code was not hit)
2) Mandatory info in bodies checked for all location, RNI and WAI API subscription, if not there, returning BadRequest error.
3) RNIS GET rab_info supporting query parameters as per spec fix : qci, ipv4 address based on name as per current implementation, cellId being an array an not a comma separated string

ONLY #3 has impact on UT, ut updated and passed
no impact on cypress.

### CODE REVIEW

**Review by dilallkx — CHANGES_REQUESTED** [review]

File: go-apps/meep-loc-serv/server/loc-serv.go [hunk 1]
```diff
@@ -105,6 +105,9 @@ var baseKey string
 var sessionMgr *sm.SessionMgr
 var mutex sync.Mutex
 
+var previousNbUsersInAp int
+var previousNbUsersInZone int
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-loc-serv/server/loc-serv.go:109]: removed previous variables, used the stored values
removed the function that just added complexity for direct calls

File: go-apps/meep-loc-serv/server/loc-serv.go [hunk 2]
```diff
@@ -357,48 +362,62 @@ func checkNotificationRegisteredZoneStatus(zoneId string, apId string, nbUsersIn
...
-						log.Error(err)
-						continue
-					}
+				//always updates the current number of users if it has changed
+				//only send notification if the number of users has changed from the last time it was sent
+				if previousNbUsersInZone != nbUsersInZone {
+					previousNbUsersInZone = nbUsersInZone
+				} else {
+					zoneWarning = false
+				}
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-loc-serv/server/loc-serv.go:371]: yes, and i merged the 2 ifs as requested, fixed

File: go-apps/meep-loc-serv/server/loc-serv.go [hunk 3]
```diff
@@ -357,48 +362,62 @@ func checkNotificationRegisteredZoneStatus(zoneId string, apId string, nbUsersIn
...
+				if nbUsersInAP >= zoneStatus.NbUsersInAPThreshold {
+					apWarning = true
+				}
+				//always updates the current number of users if it has changed
+				//only send notification if the number of users has changed from the last time it was sent
+				if previousNbUsersInAp != nbUsersInAP {
+					previousNbUsersInAp = nbUsersInAP
+				} else {
+					apWarning = false
 				}
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-loc-serv/server/loc-serv.go:389]: fixed

File: go-apps/meep-loc-serv/server/loc-serv.go [hunk 4]
```diff
@@ -1486,6 +1586,8 @@ func cleanUp() {
 	nextZonalSubscriptionIdAvailable = 1
 	nextUserSubscriptionIdAvailable = 1
 	nextZoneStatusSubscriptionIdAvailable = 1
+	previousNbUsersInAp = 0
+	previousNbUsersInZone = 0
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-loc-serv/server/loc-serv.go:1590]: fixed

**Review by dilallkx — APPROVED** [review]

**Pull request merged**